freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -20,13 +20,148 @@
 # SOFTWARE.
 #
 # Written by Valerij Maljulin <vmaljuli@redhat.com>
+# Written by Chuang Zhang <chuazhan@redhat.com>
 
-from freshmaker.handlers import ContainerBuildHandler
+import requests
+from http import HTTPStatus
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+from freshmaker import conf, db, log
+from freshmaker.errata import Errata
 from freshmaker.events import FlatpakModuleAdvisoryReadyEvent
+from freshmaker.handlers import ContainerBuildHandler, fail_event_on_handler_exception
+from freshmaker.kojiservice import koji_service
+from freshmaker.lightblue import LightBlue
+from freshmaker.models import Event
+from freshmaker.pyxis import Pyxis
+from freshmaker.types import EventState
 
 
 class RebuildFlatpakApplicationOnModuleReady(ContainerBuildHandler):
-    name = 'RebuildFlatpakApplicationOnModuleReady'
+    name = "RebuildFlatpakApplicationOnModuleReady"
 
     def can_handle(self, event):
         return isinstance(event, FlatpakModuleAdvisoryReadyEvent)
+
+    @fail_event_on_handler_exception
+    def handle(self, event):
+
+        if event.dry_run:
+            self.force_dry_run()
+
+        self.event = event
+
+        db_event = Event.get_or_create_from_event(db.session, event)
+        self.set_context(db_event)
+
+        auto_build_images_list = self._get_auto_rebuild_image_list(event)
+        rebuild_images = self._filter_images_with_higher_rpm_nvr(
+            event, auto_build_images_list
+        )
+        if not auto_build_images_list or not rebuild_images:
+            msg = "There is no image can be rebuilt. " f"message_id: {event.msg_id}"
+            db_event.transition(EventState.SKIPPED, msg)
+            db.session.commit()
+            self.log_info(msg)
+            return []
+
+        images_nvr = ",".join([rebuild_image.nvr for rebuild_image in rebuild_images])
+        self.log_info("Following images %s will be rebuilt" % images_nvr)
```

Comment:
Images get from `self._get_auto_rebuild_image_list(event)` are directly affected images, probably you should also check if there are any images need to be rebuilt as dependency, I believe it should not happen with normal flatpak applications images, but it may be possible for flatpak runtime or sdk images, so it would be safe to also check that.

The context behind this is, for example we have two images:
Image `foo`'s Dockerfile:
```
FROM xxx
RUN yum install -y titi
```
Image `bar`'s Dockerfile
```
FROM foo
RUN yum install -y tata
```
In image `foo`, there are both `titi` and `tata` installed, when there is a CVE fix for `titi`, image `bar` is impacted, for some reason, image `foo` is not enabled for auto-rebuild, so only `bar` appears in our rebuild list. However if you only rebuild `bar`, the `titi` rpm in rebuilt image `bar` won't be updated. In such cases, you need to rebuild image `foo` as dependency first, and then use the rebuilt `foo` as parent of `bar`, so there will be two batches of images to be rebuilt:
```
batch1: foo (rebuild_reason: dependency)
batch2: bar (rebuild_reason: directly_affeced)
```
You can do this in another PR.

Reply:
Maybe we can just assume that all flatpak modules are installed from the impacted images returned by flatpak index service at this moment, if that can happen we can fix it later, or you can check this with flatpak team to confirm.

Reply:
Ok. Will check with Flatpak people.

Reply:
I got the answer from Owen Taylor "The index is returning all the images that should be rebuilt - there is no docker-style image layered / derived images with Flatpaks.". So let's rebuild the images flatpak index service returned.